tfx/components/base/base_executor_test.py

```diff
@@ -37,21 +44,130 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+                    input_dict: Dict[Text, List[types.Artifact]],
+                    output_dict: Dict[Text, List[types.Artifact]],
+                    exec_properties: Dict[Text, Any]) -> None:
+    output_dir = output_dict[_SHUFFLED_EXAMPLES].uri
+    _ = (beam_outputs[_SHUFFLED_EXAMPLES]
+         | beam.io.textio.WriteToText(output_dir))
+
+  def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+         output_dict: Dict[Text, List[types.Artifact]],
+         exec_properties: Dict[Text, Any]) -> None:
+    with beam.Pipeline(runner=beam.runners.create_runner('DirectRunner')) as p:
+      beam_inputs = self.read_inputs(p, input_dict, output_dict,
+                                     exec_properties)
+      beam_outputs = self.construct_beam_graph(p, beam_inputs, input_dict,
+                                               output_dict, exec_properties)
+      self.write_outputs(p, beam_outputs, input_dict, output_dict,
+                         exec_properties)
+
 class BaseExecutorTest(tf.test.TestCase):
 
   def testBeamSettings(self):
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--runner=DirectRunner'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(StandardOptions)
+    options = executor._make_beam_pipeline().options.view_as(StandardOptions) # pylint: disable=protected-access
     self.assertEqual('DirectRunner', options.runner)
 
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--direct_num_workers=2'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(DirectOptions)
+    options = executor._make_beam_pipeline().options.view_as(DirectOptions) # pylint: disable=protected-access
     self.assertEqual(2, options.direct_num_workers)
 
 
+class BeamExecutorTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(BeamExecutorTest, self).setUp()
+    self._input_data_path = (
+        os.path.join(os.environ.get(
+            'TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()), 'input.txt'))
+    output_data_path = (
+        os.path.join(os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR',
+                                    self.get_temp_dir()), 'output.txt'))
+    self._full_output_data_path = output_data_path + '-00000-of-00001'
+
+    self._generate_data(self._input_data_path)
+
+    # Create input dict
+    examples = types.standard_artifacts.Examples()
+    examples.uri = self._input_data_path
+    self._input_dict = {_EXAMPLES: examples}
+
+    # Create output dict
+    shuffled_examples = types.standard_artifacts.Examples()
+    shuffled_examples.uri = output_data_path
+    self._output_dict = {_SHUFFLED_EXAMPLES: shuffled_examples}
+
+    # Run Do() method
+    beam_test_executor = _BeamTestExecutor()
+    beam_test_executor.Do(self._input_dict, self._output_dict, {})
```

Comment:
Move `Do` into the test methods since it won't be used in every method after addressing @charlesccychen's comments.

Reply:
Done.